tests/acceptance/bootstrap/NotificationContext.php

```diff
@@ -66,6 +67,41 @@ public function deleteDeprovisioningNotification(): void {
 		);
 	}
 
+	/**
+	 * Delete all the inbucket emails
+	 *
+	 * @return void
+	 * @throws GuzzleException
+	 */
+	public function clearInbucketMessages(): void {
+		try {
+			$usersList = $this->featureContext->getAllCreatedUsers();
+			foreach ($usersList as $emailRecipient) {
+				EmailHelper::deleteAllEmailsForAMailbox(
+					EmailHelper::getLocalEmailUrl(),
+					$this->featureContext->getStepLineRef(),
+					$emailRecipient['email']
+				);
+			}
+		} catch (Exception $e) {
+			echo __METHOD__ .
+				" could not delete inbucket messages, is inbucket set up?\n" .
+				$e->getMessage();
+		}
+	}
+
+	/**
+	 * @AfterScenario
+	 *
+	 * @return void
+	 * @throws GuzzleException
+	 * @throws JsonException
+	 */
+	public function deleteAllNotifications(): void {
```

Comment:
we should separate in-app and email notification cleanup methods.
we can add `clearEmails` and `deleteAllNotifications` hooks then we can make use of `@email` tag

Reply:
why we need email tag please describe scenario?

Reply:
we only require email server (inbucket) for email related tests. what we try to achieve was to do the email cleanup only if we are running email related tests thus preventing email cleanup for all tests.
I must add `@email` tag for all email related tests and removing the tag would be against the initial implementation.